src/tiled/mainwindow.cpp

```diff
@@ -1509,6 +1493,98 @@ void MainWindow::retranslateUi()
+    if (outputFiles.size() > 0) {
+        // Check if any output file already exists
+        QString message =
+                tr("Some export files already exist:") + QLatin1String("\n\n");
+
+        bool overwriteHappens = false;
+
+        for (const QString &outputFile : outputFiles) {
+            if (QFile::exists(outputFile)) {
+                overwriteHappens = true;
+                message += outputFile + QLatin1Char('\n');
+            }
+        }
+        message += QLatin1Char('\n') + tr("Do you want to replace them?");
+
+        // If overwrite happens, warn the user and get confirmation before exporting
+        if (overwriteHappens) {
+            const QMessageBox::StandardButton reply = QMessageBox::warning(
+                                                          this,
+                                                          tr("Overwrite Files"),
+                                                          message,
+                                                          QMessageBox::Yes | QMessageBox::No,
+                                                          QMessageBox::No);
+
+            if (reply != QMessageBox::Yes)
+                return;
+        }
+    }
+
+    Preferences *pref = Preferences::instance();
+
+    pref->setLastPath(Preferences::ExportedFile, QFileInfo(exportDetails.mFileName).path());
+    mSettings.setValue(QLatin1String("lastUsedExportFilter"), selectedFilter);
+
+    auto exportResult = exportDetails.mFormat->write(mapDocument->map(), exportDetails.mFileName);
+    if (!exportResult)
+    {
+        QMessageBox::critical(this, tr("Error Exporting Map!"),
+                              exportDetails.mFormat->errorString());
+    } else {
+        // Remember export parameters, so subsequent exports can be done faster
+        mapDocument->setLastExportFileName(exportDetails.mFileName);
+        mapDocument->setExportFormat(exportDetails.mFormat);
+    }
+}
+
+void MainWindow::exportTilesetAs(TilesetDocument *tilesetDocument)
+{
+    QString fileName = tilesetDocument->fileName();
+    QString selectedFilter =
+            mSettings.value(QLatin1String("lastUsedExportFilter")).toString();
+    auto exportDetails = chooseExportDetails<TilesetFormat>(fileName,
+                                                    tilesetDocument->lastExportFileName(),
+                                                    selectedFilter,
+                                                    this);
+    if (!exportDetails.isValid())
+        return;
+
+    Preferences *pref = Preferences::instance();
+
+    pref->setLastPath(Preferences::ExportedFile, QFileInfo(exportDetails.mFileName).path());
+    mSettings.setValue(QLatin1String("lastUsedExportFilter"), selectedFilter);
+
+    auto exportResult = exportDetails.mFormat->write(*tilesetDocument->tileset(), exportDetails.mFileName);
+    if ( ! exportResult )
+    {
```

Comment:
Coding style: please put `{` on the same line as the `if`, don't put spaces on the inside of parentheses and no space after the `!`.

Reply:
Done